GDJS/Runtime/runtimegame.ts

```diff
@@ -681,54 +594,150 @@ namespace gdjs {
+     *
+     * When a game is hot-reload, this method can be called with the current
+     * layer.
      */
-    loadAllAssetsAsync = async (
+    async loadFirstAssets(
+      firstSceneName: string,
       progressCallback?: (progress: float) => void
-    ) => {
+    ): Promise<void> {
       try {
-        const loadingScreen = new gdjs.LoadingScreenRenderer(
-          this.getRenderer(),
-          this._imageManager,
-          this._data.properties.loadingScreen
+        await this.loadAssetsWithLoadingScreen(
+          /* isFirstLayout = */ true,
+          async (onProgress) => {
+            // TODO Is a setting needed?
+            if (false) {
+              await this._resourcesLoader.loadAllResources(onProgress);
+            } else {
+              await this._resourcesLoader.loadGlobalAndFirstLayoutResources(
+                firstSceneName,
+                onProgress
+              );
+              // Don't await as it must not block the first layout from starting.
+              this._resourcesLoader.loadAllLayoutInBackground();
+            }
+          },
+          progressCallback
         );
-        const allAssetsTotal = this._data.resources.resources.length;
-        let loadedAssets = 0;
-
-        const onProgress = (count: integer, total: integer) => {
-          const percent = Math.floor(
-            (100 * (loadedAssets + count)) / allAssetsTotal
-          );
-          loadingScreen.setPercent(percent);
-          if (progressCallback) {
-            progressCallback(percent);
-          }
-        };
-
-        loadedAssets += await this._imageManager.loadTextures(onProgress);
-        loadedAssets += await this._soundManager.preloadAudio(onProgress);
-        loadedAssets += await this._fontManager.loadFonts(onProgress);
-        loadedAssets += await this._jsonManager.preloadJsons(onProgress);
-        loadedAssets += await this._model3DManager.loadModels(onProgress);
-        await this._bitmapFontManager.loadBitmapFontData(onProgress);
-        await loadingScreen.unload();
+        // TODO This is probably not necessary in case of hot reload.
         await gdjs.getAllAsynchronouslyLoadingLibraryPromise();
       } catch (e) {
         if (this._debuggerClient) this._debuggerClient.onUncaughtException(e);
 
         throw e;
       }
-    };
+    }
+
+    /**
+     * Load all assets for a given layout, displaying progress in renderer.
+     */
+    async loadLayoutAssets(
+      layoutName: string,
+      progressCallback?: (progress: float) => void
+    ): Promise<void> {
+      await this.loadAssetsWithLoadingScreen(
+        /* isFirstLayout = */ false,
+        async (onProgress) => {
+          await this._resourcesLoader.loadAndProcessLayoutResources(
+            layoutName,
+            onProgress
+          );
+        },
+        progressCallback
+      );
+    }
+
+    /**
+     * Load assets, displaying progress in renderer.
+     */
+    async loadAssetsWithLoadingScreen(
```

Comment:
Let's avoid polluting the API surface:
```suggestion
    private async _loadAssetsWithLoadingScreen(
```